code/modules/mob/living/living.dm

```diff
@@ -158,7 +158,9 @@
 	if(!client && (mob_size < MOB_SIZE_SMALL))
 		return
 	now_pushing = TRUE
-	var/t = get_dir(src, AM)
+	var/t = src.dir
+	if(src.loc != AM.loc)
+		t = get_dir(src, AM)
```

Comment:
get_dir returns 0 if the two arguments are on the same tile

Reply:
makes sense